packages/x-date-pickers/src/internals/components/PickersPopper.tsx

```diff
@@ -337,27 +377,20 @@ export function PickersPopper(inProps: PickerPopperProps) {
           disableEnforceFocus={role === 'tooltip'}
           isEnabled={() => true}
           {...TrapFocusProps}
+          {...componentsProps?.desktopTrapFocus}
         >
-          <TransitionComponent {...TransitionProps}>
-            <PickersPopperPaper
-              tabIndex={-1}
-              elevation={8}
-              ref={handlePaperRef}
+          <Transition {...TransitionProps} {...componentsProps?.desktopTransition}>
+            <Paper
+              {...paperProps}
               onClick={(event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
                 onPaperClick(event);
-                if (onPaperClickProp) {
-                  onPaperClickProp(event);
-                }
+                paperProps.onClick?.(event);
               }}
               onTouchStart={(event: React.TouchEvent<HTMLDivElement>) => {
                 onPaperTouchStart(event);
-                if (onPaperTouchStartProp) {
-                  onPaperTouchStartProp(event);
-                }
+                paperProps.onTouchStart?.(event);
               }}
               ownerState={{ ...ownerState, placement }}
-              className={classes.paper}
-              {...otherPaperProps}
             >
               <PaperContent {...componentsProps?.paperContent}>
```

Comment:
Why not merging `Paper` and `PaperContent` in a single element?

If I remember correctly, the `PaperContent` being a `Fragment` was just to avoid breaking changes. Now we are allowed to break it's the opportunity

@LukasTy Was there another reason?

Reply:
After many discussions and experimentations—let's isolate this PR only for the refactor of props -> slots.
Merging `Paper` and `PaperContent` is not trivial as `Static` pickers do not have a wrapping Paper.
And we also have the issue of possibly having difficulties changing component layout once we flatten the structure of sub-components.
I think then will be the time to revisit this topic and look into a way of having a homogenous approach towards overriding/slots.